web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -75,14 +90,37 @@
         @RequestParam(defaultValue = "ASC") Direction direction
     ) {
         String sort = sortBy == null ? null : sortBy.getOriginalValue();
-        
+        List<String> studyIds = null;
+        if (usingAuth()) {
+            /*
+             If using auth, filter the list of samples returned using the list of study ids the
+             user has access to. If the user has access to no studies, the endpoint should not 403,
+             but instead return an empty list.
+            */
+            studyIds = studyService
+                .getAllStudies(
+                    null,
+                    projection.name(),
+                    PagingConstants.MAX_PAGE_SIZE,
+                    0,
+                    null,
+                    direction.name()
+                )
+                .stream()
+                .map(CancerStudy::getCancerStudyIdentifier)
+                .collect(Collectors.toList());
+        }
+
         if (projection == Projection.META) {
             HttpHeaders httpHeaders = new HttpHeaders();
-            httpHeaders.add(HeaderKeyConstants.TOTAL_COUNT, sampleService.getMetaSamples(keyword).getTotalCount().toString());
+            httpHeaders.add(
+                HeaderKeyConstants.TOTAL_COUNT,
```

Comment:
as a line continuation indentation, I would indent this 8 rather than 4 (personal preference)